summariser/src/query/host_metrics.rs

```diff
@@ -0,0 +1,65 @@
+use std::fmt;
+
+/// Kind of host metric measurement.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub enum HostMetricMeasurement {
+    Cpu,
+    Disk,
+    Diskio,
+    Kernel,
+    Mem,
+    Net,
+    Netstat,
+    Processes,
+    Swap,
+    System,
+}
+
+impl HostMetricMeasurement {
+    /// Iter all [`HostMetricMeasurement`] variants.
+    pub fn all() -> impl Iterator<Item = Self> {
+        [
+            Self::Cpu,
+            Self::Disk,
+            Self::Diskio,
+            Self::Kernel,
+            Self::Mem,
+            Self::Net,
+            Self::Netstat,
+            Self::Processes,
+            Self::Swap,
+            Self::System,
+        ]
+        .into_iter()
+    }
+}
+
+impl fmt::Display for HostMetricMeasurement {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            HostMetricMeasurement::Cpu => write!(f, "cpu"),
+            HostMetricMeasurement::Disk => write!(f, "disk"),
+            HostMetricMeasurement::Diskio => write!(f, "diskio"),
+            HostMetricMeasurement::Kernel => write!(f, "kernel"),
+            HostMetricMeasurement::Mem => write!(f, "mem"),
+            HostMetricMeasurement::Net => write!(f, "net"),
+            HostMetricMeasurement::Netstat => write!(f, "netstat"),
+            HostMetricMeasurement::Processes => write!(f, "processes"),
+            HostMetricMeasurement::Swap => write!(f, "swap"),
+            HostMetricMeasurement::System => write!(f, "system"),
+        }
+    }
+}
+/// Get SELECT query for host metrics.
+///
+/// Given a [`HostMetricMeasurement`], it returns the select for all the fields associated to the measurement collected.
+#[inline(always)]
+pub fn host_metrics_select(measurement: HostMetricMeasurement, run_id: &str) -> String {
+    format!(
+        r#"SELECT * FROM
+        "windtunnel"."autogen"."{measurement}"
+        WHERE run_id = '{run_id}'
```

Comment:
I don't want to select here from all measures because I want to aggregate by measure.

Reply:
Should this query builder take the full run summary and limit by timestamp too? 

The host metrics are associated with the run_id which gets us back the right host metrics for the scenario but we could drop the metrics collected from a little before and a little after the scenario runs. This is where I was thinking we'd do that bit that was proposed to be done when uploading the metrics. On Nomad it won't drop much, but locally it will drop whatever amount of time it took the user to run the scenario after starting to collect host metrics and the same on shutdown.